lib/UI.js

```diff
@@ -108,17 +109,36 @@ define(function (require, exports, module) {
             // If it is, the attached preview is hidden
             // and the detached preview is opened.
             if(Resizer.isVisible("#second-pane")) {
-                Resizer.hide("#second-pane");
-                $("#first-pane").addClass("expandEditor");
-                IframeBrowser.detachPreview();
+                _detachPreview();
             }
             else {
-                Resizer.show("#second-pane");
-                $("#first-pane").removeClass("expandEditor");
+                _attachPreview();
             }
         });
     }
 
+    /**
+      * Used to hide second pane, spawn detached preview, and attach beforeunload listener
+      */
+    function _detachPreview() {
+        Resizer.hide("#second-pane");
+        $("#first-pane").addClass("expandEditor");
+        $('#liveDevButton').removeClass("liveDevButton");
+        $('#liveDevButton').addClass("liveDevButtonAlt");
+        IframeBrowser.detachPreview();
+        IframeBrowser.getDetachedPreview().addEventListener("beforeunload", function(e){
+            _attachPreview();
+        }, false);
+    }
+
+    function _attachPreview() {
+        Resizer.show("#second-pane");
+        $('#liveDevButton').removeClass("liveDevButtonAlt");
+        $('#liveDevButton').addClass("liveDevButton");
+        $("#first-pane").removeClass("expandEditor");
+        IframeBrowser.getDetachedPreview().close();
```

Comment:
NOTE: you need to check the return value of `IframeBrowser.getDetachedPreview()` because it might be null:

```js
var detachedWindow = IframeBrowser.getDetachedPreview();
if(detachedWindow) {
  detachedWindow.close();
}
```

However, I have a concern here.  It looks to me like `_attachPreview` is going to get called twice, since the `beforeunload` event is going to get triggered after you call `close()` on the window, and re-run this code.  We need to make sure this only happens once.  You might need to remove the event listener here before you call `close`